lfsapi/certs.go

```diff
@@ -20,6 +21,35 @@ func isCertVerificationDisabledForHost(c *Client, host string) bool {
 	return c.SkipSSLVerify
 }
 
+// isClientCertEnabledForHost returns whether client certificate
+// are configured for the given host
+func isClientCertEnabledForHost(c *Client, host string) bool {
+
+	_, hostSslKeyOk := c.gitEnv.Get(fmt.Sprintf("http.https://%v/.sslKey", host))
+	_, hostSslCertOk := c.gitEnv.Get(fmt.Sprintf("http.https://%v/.sslCert", host))
+	_, hostSslCaInfoOk := c.gitEnv.Get(fmt.Sprintf("http.https://%v/.sslCaInfo", host))
+
+	if hostSslKeyOk && hostSslCertOk && hostSslCaInfoOk {
+		return true
+	}
+
+	return false
+}
+
+// getClientCertForHost returns a client certificate for a specific host (which may
+// be "host:port" loaded from the gitconfig
+func getClientCertForHost(c *Client, host string) tls.Certificate {
+
+	hostSslKey, _ := c.gitEnv.Get(fmt.Sprintf("http.https://%v/.sslKey", host))
```

Comment:
Before fetching these values, do you think it would be good to call `isClientCertEnabledForHost` with the `c` and `host` args to bail out early if certificates aren't enabled for that given host?